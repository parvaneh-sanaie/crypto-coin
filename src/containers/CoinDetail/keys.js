const KEYS = {
    COIN: {
        DETAIL: `COIN_DETAIL`,
        TOP10MARKETS: `TOP_10_MARKETS`,
    },
};

export default KEYS;
