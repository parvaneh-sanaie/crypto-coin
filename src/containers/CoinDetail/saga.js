import { takeLatest, call, put } from 'redux-saga/effects';
import lodash from 'lodash';

import KEYS from './keys';
import GET from '../../constants';
import request from '../../utils/request';
import {
    crudGetOneFailureAction,
    crudGetOneSuccessAction,
    crudGetAllFailureAction,
    crudGetAllSuccessAction,
} from '../../actions/crud/get';

function* getCoinDetail(action) {
    try {
        const { id } = action.payload;
        const URL = `https://api.coinlore.net/api/ticker/?id=${id}`;
        const result = yield call(request, URL);
        if (result.error) {
            yield put(crudGetOneFailureAction(action.key, { error: result.error }));
        } else if (result.data) {
            yield put(crudGetOneSuccessAction(action.key, { data: result.data[0], success: true }));
        }
    } catch (error) {
        yield put(crudGetOneFailureAction(action.key, { error }));
    }
}

function* getTop10MarketsForCoin(action) {
    try {
        const { id } = action.payload;
        const URL = `https://api.coinlore.net/api/coin/markets/?id=${id}`;
        const result = yield call(request, URL);
        if (result.error) {
            yield put(crudGetAllFailureAction(action.key, { error: result.error }));
        } else if (result.data) {
            const data = lodash.orderBy(result.data, ['volume'], ['desc']).slice(0, 10);
            yield put(crudGetAllSuccessAction(action.key, { data, success: true }));
        }
    } catch (error) {
        yield put(crudGetAllFailureAction(action.key, { error }));
    }
}

export function* getCoinDetailSaga() {
    yield takeLatest(`${GET.ONE.KEY}${KEYS.COIN.DETAIL}`, getCoinDetail);
}

export function* getTop10MarketsForCoinSaga() {
    yield takeLatest(`${GET.ALL.KEY}${KEYS.COIN.TOP10MARKETS}`, getTop10MarketsForCoin);
}
