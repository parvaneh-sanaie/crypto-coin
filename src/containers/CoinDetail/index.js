import React, { useEffect, useState } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import { useParams } from 'react-router-dom';
import { faChartSimple } from '@fortawesome/free-solid-svg-icons/faChartSimple';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faDollar } from '@fortawesome/free-solid-svg-icons/faDollar';

import KEYS from './keys';

import Header from '../../components/Header';
import BarChart from '../../components/BarChart';
import crudGetReducer from '../../reducers/crud/get';
import { InjectReducer, InjectSaga } from '../../utils';
import useCrudGet from '../../hooks/crud/get/useCrudGet';
import { getCoinDetailSaga, getTop10MarketsForCoinSaga } from './saga';

const graphConfig = [
    {
        name: 'Volume',
        dataKey: 'volume',
        fill: '#2347F3',
        stack: '0',
    },
];

function CoinDetail() {
    const { coinId } = useParams();

    const getCoinDetailsReducer = crudGetReducer(KEYS.COIN.DETAIL);
    InjectReducer({ key: KEYS.COIN.DETAIL, reducer: getCoinDetailsReducer });
    InjectSaga({ key: KEYS.COIN.DETAIL, saga: getCoinDetailSaga });

    const getTop10MarketsReducer = crudGetReducer(KEYS.COIN.TOP10MARKETS);
    InjectReducer({ key: KEYS.COIN.TOP10MARKETS, reducer: getTop10MarketsReducer });
    InjectSaga({ key: KEYS.COIN.TOP10MARKETS, saga: getTop10MarketsForCoinSaga });

    const {
        crudGetOne: getCoinDetail,
        crudGetOnePayload,
        crudGetOneLoading,
        crudGetOneSuccess,
    } = useCrudGet(KEYS.COIN.DETAIL);

    const {
        crudGetAll: getTop10MarketByCoinId,
        crudGetAllPayload,
        crudGetAllLoading,
        crudGetAllSuccess,
    } = useCrudGet(KEYS.COIN.TOP10MARKETS);

    const [top10Markets, setTop10Markets] = useState(null);
    const [coinDetail, setCoinDetail] = useState(null);

    useEffect(() => {
        if (crudGetOnePayload == null && crudGetAllPayload == null && !crudGetAllLoading && !crudGetOneLoading) {
            getCoinDetail({ id: coinId });
            getTop10MarketByCoinId({ id: coinId });
        }
    }, [
        coinId,
        crudGetAllLoading,
        crudGetAllPayload,
        crudGetOneLoading,
        crudGetOnePayload,
        getCoinDetail,
        getTop10MarketByCoinId,
    ]);

    useEffect(() => {
        if (crudGetOneSuccess) {
            setCoinDetail(crudGetOnePayload.data);
        }
    }, [crudGetOnePayload, crudGetOneSuccess]);

    useEffect(() => {
        if (crudGetAllSuccess) {
            setTop10Markets(crudGetAllPayload.data);
        }
    }, [crudGetAllPayload, crudGetAllSuccess]);

    return (
        <div data-testid="CoinDetail">
            <Header showProgress={crudGetOneLoading || crudGetAllLoading} />
            <div data-testid="container" className="container-fluid px-4 small">
                {coinDetail && top10Markets && (
                    <div className="row">
                        <div className="col-xl-7 col-md-6 mt-4">
                            <div data-testid="TitleCard" className="row card card-body flex-row">
                                <div className="col-xl-10 col-md-8">
                                    <img
                                        className="px-1 rounded-circle"
                                        src={`https://www.coinlore.com/img/25x25/${coinDetail.nameid}.png`}
                                        alt={coinDetail.symbol}
                                    />
                                    <strong data-testid="coin_symbol">{coinDetail.symbol}</strong> Price / Everything
                                    About <strong data-testid="coin_name">{coinDetail.name}</strong>
                                </div>
                                <div className="text-end align-self-end col-xl-2 col-md-4">
                                    Rank:{' '}
                                    <div
                                        data-testid="coin_rank"
                                        className={coinDetail.rank <= 10 ? `badge bg-success` : `badge bg-primary`}
                                    >
                                        {coinDetail.rank}
                                    </div>
                                </div>
                            </div>
                            <div className="card row mt-4">
                                <div className="card-body table-responsive">
                                    <table className="table table-sm">
                                        <thead>
                                            <tr>
                                                <th>Market Cap</th>
                                                <th>Vol (24H)</th>
                                                <th>Circulating Supply</th>
                                                <th>Total Supply</th>
                                            </tr>
                                        </thead>
                                        <tbody>
                                            <tr>
                                                <td>${coinDetail.market_cap_usd}</td>
                                                <td>
                                                    <div>${coinDetail.volume24}</div>
                                                    <div className="text-secondary small">
                                                        {coinDetail.volume24_native}
                                                        <strong> {coinDetail.symbol}</strong>
                                                    </div>
                                                </td>
                                                <td data-testid="csupply">${coinDetail.csupply}</td>
                                                <td data-testid="tsupply">${coinDetail.tsupply}</td>
                                            </tr>
                                        </tbody>
                                    </table>
                                </div>
                            </div>
                            <div className="row d-flex align-content-between">
                                <div className="col-xl-3 col-md-6 mt-4 pe-1 ps-0">
                                    <div className="card">
                                        <div className="card-body">
                                            <div className="card-header">
                                                <FontAwesomeIcon icon={faDollar} className="me-1" />
                                                <strong>Price</strong>
                                            </div>
                                            <div className="list-group-flush">
                                                <div className="list-group-item">
                                                    <span>USD: </span>
                                                    <span data-testid="price_usd" className="text-primary small">
                                                        ${coinDetail.price_usd}
                                                    </span>
                                                </div>
                                                <div className="list-group-item">
                                                    <span>BTC: </span>
                                                    <span data-testid="price_btc" className="text-primary small">
                                                        {coinDetail.price_btc} <small>(BTC)</small>
                                                    </span>
                                                </div>
                                            </div>
                                        </div>
                                    </div>
                                </div>
                                <div className="card card-body col-xl-9 col-md-12 mt-4">
                                    <table className="table table-striped table-bordered table-hover table-sm">
                                        <thead>
                                            <tr>
                                                <th>VS</th>
                                                <th>1h</th>
                                                <th>24h</th>
                                                <th>7d</th>
                                            </tr>
                                        </thead>
                                        <tbody>
                                            <tr>
                                                <td>USD</td>
                                                <td
                                                    data-testid="percent_change_1h"
                                                    className={
                                                        coinDetail.percent_change_1h > 0
                                                            ? `text-success`
                                                            : `text-danger`
                                                    }
                                                >
                                                    {coinDetail.percent_change_1h}%
                                                </td>
                                                <td
                                                    data-testid="percent_change_24h"
                                                    className={
                                                        coinDetail.percent_change_24h > 0
                                                            ? `text-success`
                                                            : `text-danger`
                                                    }
                                                >
                                                    {coinDetail.percent_change_24h}%
                                                </td>
                                                <td
                                                    data-testid="percent_change_7d"
                                                    className={
                                                        coinDetail.percent_change_7d > 0
                                                            ? `text-success`
                                                            : `text-danger`
                                                    }
                                                >
                                                    {coinDetail.percent_change_7d}%
                                                </td>
                                            </tr>
                                        </tbody>
                                    </table>
                                </div>
                            </div>
                        </div>
                        <div className="col-xl-5 col-md-6 mt-4">
                            <div className="card mb-4">
                                <div className="card-header">
                                    <FontAwesomeIcon icon={faChartSimple} className="me-1" />
                                    Top 10 Markets for {coinDetail.name}({coinDetail.symbol})
                                </div>
                                <div data-testid="top10Markets" className="card-body flex-row pt-4">
                                    <BarChart
                                        width={600}
                                        height={268}
                                        data={top10Markets}
                                        barConfig={graphConfig}
                                        tooltip
                                        xAxisDataKey="name"
                                    />
                                </div>
                            </div>
                        </div>
                    </div>
                )}
            </div>
        </div>
    );
}
export default CoinDetail;
