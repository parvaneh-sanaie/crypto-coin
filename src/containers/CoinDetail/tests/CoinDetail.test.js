import React from 'react';
import { cleanup, render, act, screen } from '@testing-library/react';

import CoinDetail from '../index';
import { coinId, mockTop10Markets, mockCoinDetail } from './testData';

let mockUseCrudGetReturnValue = {
    crudGetOne: jest.fn(),
    crudGetOnePayload: null,
    crudGetOneLoading: false,
    crudGetOneSuccess: false,
    crudGetAll: jest.fn(),
    crudGetAllPayload: null,
    crudGetAllLoading: false,
    crudGetAllSuccess: false,
};

jest.mock('react-redux');
jest.mock('../saga');
jest.mock('../../../components/BarChart');
jest.mock('../../../hooks/crud/get/useCrudGet', () => ({
    __esModule: true,
    default: () => mockUseCrudGetReturnValue,
}));

jest.mock('react-resize-detector', () => ({
    __esModule: true,
    default: () =>
        jest.fn().mockImplementation(() => ({
            observe: jest.fn(),
            unobserve: jest.fn(),
            disconnect: jest.fn(),
        })),
}));

jest.mock('../../../utils', () => ({
    InjectReducer: jest.fn(),
    InjectSaga: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
    useParams: () => ({ coinId: 90 }),
}));

describe('<CoinDetail />', () => {
    beforeEach(() => {
        mockUseCrudGetReturnValue = {
            crudGetOne: jest.fn(),
            crudGetOnePayload: null,
            crudGetOneLoading: false,
            crudGetOneSuccess: false,
            crudGetAll: jest.fn(),
            crudGetAllPayload: null,
            crudGetAllLoading: false,
            crudGetAllSuccess: false,
        };
    });

    it('Should be match to the snapshot', () => {
        let renderResult = null;
        act(() => {
            renderResult = render(<CoinDetail />);
        });
        expect(renderResult).toMatchSnapshot();
    });

    it('Should render an empty page', () => {
        expect.assertions(1);
        act(() => {
            render(<CoinDetail />);
        });
        const container = screen.getByTestId('container');
        expect(container).toBeInTheDocument();
    });

    it('Should display progress bar when coin detail is loading', () => {
        expect.assertions(1);
        mockUseCrudGetReturnValue.crudGetOneLoading = true;
        mockUseCrudGetReturnValue.crudGetOnePayload = { id: coinId };
        act(() => {
            render(<CoinDetail />);
        });
        expect(screen.getByTestId('progressBar')).toBeInTheDocument();
    });

    it('Should display progress bar when top 10 markets are loading', () => {
        expect.assertions(1);
        mockUseCrudGetReturnValue.crudGetAllLoading = true;
        mockUseCrudGetReturnValue.crudGetAllPayload = { id: coinId };
        act(() => {
            render(<CoinDetail />);
        });
        expect(screen.getByTestId('progressBar')).toBeInTheDocument();
    });

    it('Should display coin details in the page after page successfully loaded', () => {
        expect.assertions(12);
        mockUseCrudGetReturnValue.crudGetOnePayload = { data: mockCoinDetail };
        mockUseCrudGetReturnValue.crudGetAllPayload = { data: mockTop10Markets };
        mockUseCrudGetReturnValue.crudGetOneSuccess = true;
        mockUseCrudGetReturnValue.crudGetAllSuccess = true;
        act(() => {
            render(<CoinDetail />);
        });
        expect(screen.getByTestId('TitleCard')).toBeInTheDocument();
        expect(screen.getByTestId('coin_name').innerHTML).toContain(`${mockCoinDetail.name}`);
        expect(screen.getByTestId('coin_symbol').innerHTML).toContain(`${mockCoinDetail.symbol}`);
        expect(screen.getByTestId('coin_rank').innerHTML).toBe(`${mockCoinDetail.rank}`);
        expect(screen.getByTestId('csupply').innerHTML).toContain(`${mockCoinDetail.csupply}`);
        expect(screen.getByTestId('tsupply').innerHTML).toContain(`${mockCoinDetail.tsupply}`);
        expect(screen.getByTestId('price_usd').innerHTML).toContain(`${mockCoinDetail.price_usd}`);
        expect(screen.getByTestId('price_btc').innerHTML).toContain(`${mockCoinDetail.price_btc}`);
        expect(screen.getByTestId('percent_change_1h').innerHTML).toContain(`${mockCoinDetail.percent_change_1h}`);
        expect(screen.getByTestId('percent_change_24h').innerHTML).toContain(`${mockCoinDetail.percent_change_24h}`);
        expect(screen.getByTestId('percent_change_7d').innerHTML).toContain(`${mockCoinDetail.percent_change_7d}`);
        expect(screen.getByTestId('top10Markets')).toBeInTheDocument();
    });

    afterAll(cleanup);
});
