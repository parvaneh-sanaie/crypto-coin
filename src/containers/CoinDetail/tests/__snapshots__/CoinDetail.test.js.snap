// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CoinDetail /> Should be match to the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="CoinDetail"
      >
        <div
          data-testid="Header"
        >
          <nav
            class="navbar navbar-expand navbar-dark bg-dark"
            data-testid="navbar"
          >
            <a
              class="navbar-brand ps-3"
              data-testid="homeLink"
              href="/"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-house fa-fw me-2"
                data-icon="house"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M575.8 255.5c0 18-15 32.1-32 32.1h-32l.7 160.2c0 2.7-.2 5.4-.5 8.1V472c0 22.1-17.9 40-40 40H456c-1.1 0-2.2 0-3.3-.1c-1.4 .1-2.8 .1-4.2 .1H416 392c-22.1 0-40-17.9-40-40V448 384c0-17.7-14.3-32-32-32H256c-17.7 0-32 14.3-32 32v64 24c0 22.1-17.9 40-40 40H160 128.1c-1.5 0-3-.1-4.5-.2c-1.2 .1-2.4 .2-3.6 .2H104c-22.1 0-40-17.9-40-40V360c0-.9 0-1.9 .1-2.8V287.6H32c-18 0-32-14-32-32.1c0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7L564.8 231.5c8 7 12 15 11 24z"
                  fill="currentColor"
                />
              </svg>
              Crypto Coin
            </a>
            <form
              class="d-none d-md-inline-block form-inline ms-auto me-0 me-md-3 my-2 my-md-0"
              data-testid="search-field-form"
            >
              <div
                class="input-group"
              >
                <input
                  aria-describedby="btnNavbarSearch"
                  aria-label="Search for..."
                  class="form-control"
                  data-testid="search-field-input"
                  placeholder="Search for..."
                  type="text"
                  value=""
                />
                <button
                  class="btn btn-primary"
                  data-testid="search-field-button"
                  id="btnNavbarSearch"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-magnifying-glass "
                    data-icon="magnifying-glass"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352c79.5 0 144-64.5 144-144s-64.5-144-144-144S64 128.5 64 208s64.5 144 144 144z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </div>
            </form>
            <ul
              class="navbar-nav ms-auto ms-md-0 me-3 me-lg-4"
              data-testid="userProfileLink"
            >
              <li
                class="nav-item dropdown"
              >
                <a
                  aria-expanded="false"
                  class="nav-link dropdown-toggle"
                  data-bs-toggle="dropdown"
                  href="#"
                  role="button"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-user fa-fw"
                    data-icon="user"
                    data-prefix="far"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M272 304h-96C78.8 304 0 382.8 0 480c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32C448 382.8 369.2 304 272 304zM48.99 464C56.89 400.9 110.8 352 176 352h96c65.16 0 119.1 48.95 127 112H48.99zM224 256c70.69 0 128-57.31 128-128c0-70.69-57.31-128-128-128S96 57.31 96 128C96 198.7 153.3 256 224 256zM224 48c44.11 0 80 35.89 80 80c0 44.11-35.89 80-80 80S144 172.1 144 128C144 83.89 179.9 48 224 48z"
                      fill="currentColor"
                    />
                  </svg>
                </a>
              </li>
            </ul>
          </nav>
        </div>
        <div
          class="container-fluid px-4 small"
          data-testid="container"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="CoinDetail"
    >
      <div
        data-testid="Header"
      >
        <nav
          class="navbar navbar-expand navbar-dark bg-dark"
          data-testid="navbar"
        >
          <a
            class="navbar-brand ps-3"
            data-testid="homeLink"
            href="/"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-house fa-fw me-2"
              data-icon="house"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M575.8 255.5c0 18-15 32.1-32 32.1h-32l.7 160.2c0 2.7-.2 5.4-.5 8.1V472c0 22.1-17.9 40-40 40H456c-1.1 0-2.2 0-3.3-.1c-1.4 .1-2.8 .1-4.2 .1H416 392c-22.1 0-40-17.9-40-40V448 384c0-17.7-14.3-32-32-32H256c-17.7 0-32 14.3-32 32v64 24c0 22.1-17.9 40-40 40H160 128.1c-1.5 0-3-.1-4.5-.2c-1.2 .1-2.4 .2-3.6 .2H104c-22.1 0-40-17.9-40-40V360c0-.9 0-1.9 .1-2.8V287.6H32c-18 0-32-14-32-32.1c0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7L564.8 231.5c8 7 12 15 11 24z"
                fill="currentColor"
              />
            </svg>
            Crypto Coin
          </a>
          <form
            class="d-none d-md-inline-block form-inline ms-auto me-0 me-md-3 my-2 my-md-0"
            data-testid="search-field-form"
          >
            <div
              class="input-group"
            >
              <input
                aria-describedby="btnNavbarSearch"
                aria-label="Search for..."
                class="form-control"
                data-testid="search-field-input"
                placeholder="Search for..."
                type="text"
                value=""
              />
              <button
                class="btn btn-primary"
                data-testid="search-field-button"
                id="btnNavbarSearch"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-magnifying-glass "
                  data-icon="magnifying-glass"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352c79.5 0 144-64.5 144-144s-64.5-144-144-144S64 128.5 64 208s64.5 144 144 144z"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </div>
          </form>
          <ul
            class="navbar-nav ms-auto ms-md-0 me-3 me-lg-4"
            data-testid="userProfileLink"
          >
            <li
              class="nav-item dropdown"
            >
              <a
                aria-expanded="false"
                class="nav-link dropdown-toggle"
                data-bs-toggle="dropdown"
                href="#"
                role="button"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-user fa-fw"
                  data-icon="user"
                  data-prefix="far"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M272 304h-96C78.8 304 0 382.8 0 480c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32C448 382.8 369.2 304 272 304zM48.99 464C56.89 400.9 110.8 352 176 352h96c65.16 0 119.1 48.95 127 112H48.99zM224 256c70.69 0 128-57.31 128-128c0-70.69-57.31-128-128-128S96 57.31 96 128C96 198.7 153.3 256 224 256zM224 48c44.11 0 80 35.89 80 80c0 44.11-35.89 80-80 80S144 172.1 144 128C144 83.89 179.9 48 224 48z"
                    fill="currentColor"
                  />
                </svg>
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div
        class="container-fluid px-4 small"
        data-testid="container"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
